tardis/__init__.py

```diff
@@ -33,6 +33,125 @@
 logging.getLogger("py.warnings").addHandler(console_handler)
 
+
+class FilterLog(object):
+    """
+    Filter Log Class for Filtering Logging Output
+    to a particular level
+
+    Parameters
+    ----------
+    log_level : logging object
+        allows to have a filter for the
+        particular log_level
+    """
+
+    def __init__(self, log_level):
+        self.log_level = log_level
+
+    def filter(self, log_record):
+        """
+        filter() allows to set the logging level for
+        all the record that are being parsed & hence remove those
+        which are not of the particular level
+
+        Parameters
+        ----------
+        log_record : logging.record
+            which the paricular record upon which the
+            filter will be applied
+
+        Returns
+        -------
+        boolean : True, if the current log_record has the
+            level that of the specified log_level
+            False, if the current log_record doesn't have the
+            same log_level as the specified one
+        """
+        return log_record.levelno == self.log_level
+
+
+list_of_filter = []
+
+
+def logging_state(log_state, tardis_config, specific):
+    """
+    Function to set the logging configuration for the simulation output
+    Called from within run_tardis()
+    Configured via functional arguments passed through run_tardis() - log_state & specific
+    Configured via YAML parameters under `debug` section - logging_level & specific_logging
+
+    Parameters
+    ----------
+    log_state: str
+        Allows to input the log level for the simulation
+        Uses Python logging framework to determine the messages that will be output
+    specific: boolean
+        Allows to set specific logging levels. Logs of the `log_state` level would be output.
+    """
+    if "debug" in tardis_config:
+        if specific or tardis_config["debug"]["specific_logging"]:
+            specific = True
+        else:
+            specific = False
+
+        if tardis_config["debug"]["logging_level"] or log_state:
+            if (
+                log_state.upper() == "CRITICAL"
+                and tardis_config["debug"]["logging_level"]
+            ):
+                logging_level = tardis_config["debug"]["logging_level"]
+            elif log_state:
+                logging_level = log_state
+                if tardis_config["debug"]["logging_level"] and log_state:
+                    print("log_state & logging_level both specified")
+                    print(
+                        f"log_state = {log_state} will be used for Log Level Determination\n"
+                    )
+            else:
+                logging_level = tardis_config["debug"]["logging_level"]
+    else:
+        logging_level = log_state
+        specific = specific
+
+    loggers = [
+        logging.getLogger(name) for name in logging.root.manager.loggerDict
+    ]
+    if logging_level.upper() in [
+        "NOTSET",
```

Comment:
this is a constant. please define it somewhere.

Reply:
this is repeated and should defined as a CONSTANT above and then you can check both.

Reply:
Yes, Sure. That makes a lot of sense. 😄